tests/sentry/auth/test_helper.py

```diff
@@ -309,6 +308,25 @@ def test_authenticated(self, mock_render):
         assert context["existing_user"] is self.request.user
         assert "login_form" not in context
 
+    @mock.patch("sentry.auth.helper.render_to_response")
+    @mock.patch("sentry.auth.helper.create_verification_key")
+    def test_unauthenticated_with_existing_user(self, mock_create_key, mock_render):
```

Comment:
it seems like either we are going to end up having duplicated test behavior between `tests/sentry/web/frontend/test_auth_organization_login.py` and here, or have tested behavior be split across both. i'm not sure what the right resolution is, but I think the integration tests are easier to work with since you don't need so many mocks.

Reply:
as discussed offline, probably prudent to enable this flag in the org login tests to confirm existing behavior is preserved when this flag is enabled.

Reply:
@RyanSkonnord what was your conclusion on integration tests vs unit tests? It seems like we'd still want integration tests regardless

Reply:
Added an integration test: https://github.com/getsentry/sentry/pull/28621/commits/833a63361e6518638d3acd78a2561354058dd53b

See also https://github.com/getsentry/sentry/pull/28784 😅